packages/block-editor/src/components/block-tools/block-selection-button.js

```diff
@@ -190,7 +190,7 @@ function BlockSelectionButton( { clientId, rootClientId, blockElement } ) {
 			if ( focusedBlockUid ) {
 				event.preventDefault();
 				selectBlock( focusedBlockUid );
-			} else if ( isTab && selectedBlockClientId ) {
+			}/* else if ( isTab && selectedBlockClientId ) {
```

Comment:
What is this line for? It seems to not be working properly. Even checked out on trunk this line is an issue. If I have one Paragraph and one Columns, I will enter Gutenberg Navigation Mode. If I press Tab, I will land on Paragraph, then Columns, then I get the first button within Columns. After commenting this line, I get placed in the Title field and I can go through my list again which seems like a much better workflow than random focus of Columns block content and possibly other Blocks.

Reply:
I am trying to fix this in the below PR.

https://github.com/WordPress/gutenberg/pull/37965